--- a/fs/stat.c
+++ b/fs/stat.c
@@ -177,6 +177,7 @@
 	struct path path;
 	unsigned lookup_flags = 0;
 	int error;
+	extern bool ghost_hidden_path(const struct path *gp);
 
 	if (flags & ~(AT_SYMLINK_NOFOLLOW | AT_NO_AUTOMOUNT | AT_EMPTY_PATH |
 		      AT_STATX_SYNC_TYPE))
@@ -193,6 +194,11 @@
 	error = user_path_at(dfd, filename, lookup_flags, &path);
 	if (error)
 		goto out;
+	if (ghost_hidden_path(&path)) {
+		path_put(&path);
+		error = -ENOENT;
+		goto out;
+	}
 
 	error = vfs_getattr(&path, stat, request_mask, flags);
 	stat->mnt_id = real_mount(path.mnt)->mnt_id;
