--- a/fs/namei.c
+++ b/fs/namei.c
@@ -4242,6 +4242,7 @@
 	struct inode *delegated_inode = NULL;
 	int how = 0;
 	int error;
+	extern bool ghost_hidden_path(const struct path *gp);
 
 	if ((flags & ~(AT_SYMLINK_FOLLOW | AT_EMPTY_PATH)) != 0)
 		return -EINVAL;
@@ -4262,6 +4263,10 @@
 	error = user_path_at(olddfd, oldname, how, &old_path);
 	if (error)
 		return error;
+	if (ghost_hidden_path(&old_path)) {
+		path_put(&old_path);
+		return -ENOENT;
+	}
 
 	new_dentry = user_path_create(newdfd, newname, &new_path,
 					(how & LOOKUP_REVAL));
