--- a/fs/namei.c
+++ b/fs/namei.c
@@ -4769,6 +4769,7 @@
 	struct inode *delegated_inode = NULL;
 	int how = 0;
 	int error;
+	extern bool ghost_hidden_path(const struct path *gp);
 
 	if ((flags & ~(AT_SYMLINK_FOLLOW | AT_EMPTY_PATH)) != 0) {
 		error = -EINVAL;
@@ -4789,6 +4790,10 @@
 	error = filename_lookup(olddfd, old, how, &old_path, NULL);
 	if (error)
 		goto out_putnames;
+	if (ghost_hidden_path(&old_path)) {
+		error = -ENOENT;
+		goto out_putpath;
+	}
 
 	new_dentry = filename_create(newdfd, new, &new_path,
 					(how & LOOKUP_REVAL));
