--- a/fs/proc/task_mmu.c
+++ b/fs/proc/task_mmu.c
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 #include <linux/pagewalk.h>
 #include <linux/mm_inline.h>
+#include <linux/ksu.h>
 #include <linux/hugetlb.h>
 #include <linux/huge_mm.h>
 #include <linux/mount.h>
@@ -317,6 +318,9 @@ show_map_vma(struct seq_file *m, struct vm_area_struct *vma)
 	unsigned long ino = 0;
 	dev_t dev = 0;
 	const char *name = NULL;
+
+	if (ksu_is_hidden_uid(current_uid().val))
+		return;
 
 	if (file) {
 		struct inode *inode = file_inode(vma->vm_file);
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -95,6 +95,7 @@
 #include <linux/time_namespace.h>
 #include <linux/resctrl.h>
 #include <linux/cn_proc.h>
+#include <linux/ksu.h>
 #include <trace/events/oom.h>
 #include "internal.h"
 #include "fd.h"
@@ -3100,6 +3101,9 @@ static int do_task_stat(struct seq_file *m, struct pid_namespace *ns,
 	u64 cutime, cstime, utime, stime;
 	unsigned long cmin_flt = 0, cmaj_flt = 0;
 	unsigned long  min_flt = 0,  maj_flt = 0;
+
+	if (ksu_is_hidden_uid(current_uid().val))
+		return -ENOENT;
 
 	state = *get_task_state(task);
 	vsize = eip = esp = 0;
