--- ./kernel/apk_sign.c	2025-05-04 23:17:31.686152813 -0400
+++ ./kernel/apk_sign.c	2025-05-04 23:20:30.307512274 -0400
@@ -316,5 +316,10 @@
 
 bool ksu_is_manager_apk(char *path)
 {
-	return check_v2_signature(path, EXPECTED_NEXT_SIZE, EXPECTED_NEXT_HASH);
-}
\ No newline at end of file
+	return (check_v2_signature(path, EXPECTED_NEXT_SIZE)
+	|| check_v2_signature(path, 384) // 5ec1cff
+	|| check_v2_signature(path, 0x3e6) // ksu-next
+	|| check_v2_signature(path, 0x396) // rsuntk
+	|| check_v2_signature(path, 0x35c) // ShirkNeko udochina
+	|| check_v2_signature(path, 0x363) // backslashxx
+	);
+}
