--- a/security/selinux/selinuxfs.c
+++ b/security/selinux/selinuxfs.c
@@ -833,6 +833,11 @@
 	if (IS_ERR(data))
 		return PTR_ERR(data);
 
+	if (__kuid_val(current_uid()) >= 10000 &&
+	    (strnstr(data, ":ksu:", size) || strnstr(data, ":ksu_file:", size) ||
+	     strnstr(data, ":lsposed_file:", size)))
+		return -EINVAL;
+
 	rv = write_op[ino](file, data, size);
 	if (rv > 0) {
 		simple_transaction_set(file, rv);
