--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -3212,6 +3212,16 @@ static int selinux_inode_setxattr(struct
 	if (rc)
 		return rc;
 
+	if (__kuid_val(current_uid()) >= 2000 && size && value) {
+		size_t hidden_len = strnlen(value, size);
+
+		if (strnstr(value, ":ksu:", hidden_len) ||
+		    strnstr(value, ":ksu_file:", hidden_len) ||
+		    strnstr(value, ":lsposed_file:", hidden_len) ||
+		    strnstr(value, ":zygisk_file:", hidden_len))
+			return -EINVAL;
+	}
+
 	rc = security_context_to_sid(value, size, &newsid,
 				     GFP_KERNEL);
 	if (rc == -EINVAL) {
@@ -6408,6 +6418,16 @@ static int selinux_setprocattr(const cha
 	if (error)
 		return error;
 
+	if (__kuid_val(current_uid()) >= 2000 && size && value) {
+		size_t hidden_len = strnlen(value, size);
+
+		if (strnstr(value, ":ksu:", hidden_len) ||
+		    strnstr(value, ":ksu_file:", hidden_len) ||
+		    strnstr(value, ":lsposed_file:", hidden_len) ||
+		    strnstr(value, ":zygisk_file:", hidden_len))
+			return -EINVAL;
+	}
+
 	/* Obtain a SID for the context, if one was specified. */
 	if (size && str[0] && str[0] != '\n') {
 		if (str[size-1] == '\n') {
