--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -3256,6 +3256,16 @@ static int selinux_inode_setxattr(struct
 	if (rc)
 		return rc;
 
+	if (__kuid_val(current_uid()) >= 2000 && size && value) {
+		size_t hidden_len = strnlen(value, size);
+
+		if (strnstr(value, ":ksu:", hidden_len) ||
+		    strnstr(value, ":ksu_file:", hidden_len) ||
+		    strnstr(value, ":lsposed_file:", hidden_len) ||
+		    strnstr(value, ":zygisk_file:", hidden_len))
+			return -EINVAL;
+	}
+
 	rc = security_context_to_sid(value, size, &newsid,
 				     GFP_KERNEL);
 	if (rc == -EINVAL) {
@@ -6462,6 +6472,16 @@ static int selinux_lsm_setattr(u64 attr,
 			str[size-1] = 0;
 			size--;
 		}
+		{
+			size_t hidden_len = strnlen(value, size);
+
+			if (__kuid_val(current_uid()) >= 2000 &&
+			    (strnstr(value, ":ksu:", hidden_len) ||
+			     strnstr(value, ":ksu_file:", hidden_len) ||
+			     strnstr(value, ":lsposed_file:", hidden_len) ||
+			     strnstr(value, ":zygisk_file:", hidden_len)))
+				return -EINVAL;
+		}
 		error = security_context_to_sid(value, size,
 						&sid, GFP_KERNEL);
 		if (error == -EINVAL && attr == LSM_ATTR_FSCREATE) {
